dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/NonPersistedDataApi.kt

```diff
@@ -0,0 +1,36 @@
+package org.dataland.datalandbackend.api
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+
+/**
+ * Defines the restful dataland-backend API regarding internal data exchange
+ */
+@RequestMapping("/internal/nonpersisted")
+interface NonPersistedDataApi {
```

Comment:
I don't like the name NonPersisted.

We should always see things from the point of view of the backend: The backend does not know that there are things that should be persisted. It just knows: I received a dataset that I will provide for some time until something happens.

Reply:
This view should be reflected in the classname. Open for suggestions.

Reply:
Maybe something like CachedData?

Reply:
DataDeliveryApi?

Reply:
`RecievedData`?

Reply:
Let's have a team discussion tomorrow.

Reply:
But the backend also knows that it is not going to keep that data and the implementation actually deletes it in the moment the data is requested the first time. The suggested names all do not clarify this circumstance.

Reply:
I now went with TemporarilyCachedDataApi (and so on) and the endpoint is reachible under /api/internal/cached/{dataId}

Reply:
ok